### PR TITLE
add ilo support higher than ilo4

### PR DESCRIPTION
**Pull request by FLiPp3r90**
Issue https://github.com/CSCfi/ansible-role-hp-firmware-upgrade/issues/23

This isn't tested with ILO5 but works with ILO4.

### CODE REVIEW

File: tasks/prerequisite.yml [hunk 1]
```diff
@@ -25,4 +25,5 @@
   package: name={{ item }} state=latest
   with_items:
     - hp-scripting-tools
+    - hponcfg
```
Comment (martbhell) [inline tasks/prerequisite.yml:28]: We do not have hponcfg in the repo we use (fwpp [1]).

So this brings in a dependency on another repo for us, which I'm not a big fan of. Especially I don't want to mix hpspp and fwpp repos. Do you know if hponcfg is in some other of their repos?

[1]: https://downloads.linux.hpe.com/SDR/project/fwpp/index.html

File: tasks/ilo.yml [hunk 2]
```diff
@@ -1,34 +1,33 @@
...
 
-   # We look for:
-   # 01:00.0 "System peripheral" "Hewlett-Packard Company" "Integrated Lights-Out Standard Slave Instrumentation & System Support" -r05 "Hewlett-Packard Company" "iLO4"
- - name: Checking for ILO device
-   command: lspci -s 01:00.0 -m
-   register: ilo_info
-   changed_when: False
-   check_mode: no
+- name: Checking for ILO device
+  shell: hponcfg -w conf.txt|grep -i 'Firmware Revision'| awk '{print $8$9}'
```
Comment (martbhell) [inline tasks/ilo.yml:7]: Turns out our company has some HPE Gen10 servers! 

The output from the lspci command is this:

<pre>
01:00.0 "System peripheral" "Hewlett-Packard Company" "Integrated Lights-Out Standard Slave Instrumentation & System Support" -r07 "Hewlett Packard Enterprise" "iLO5"`
</pre>

Which based on my awesome bash hacking should output "iLO5"

<pre>
$ echo '01:00.0 "System peripheral" "Hewlett-Packard Company" "Integrated Lights-Out Standard Slave Instrumentation & System Support" -r06 "Hewlett-Packard Company" "iLO4"'|rev|cut -d '"' -f2|rev
iLO5
</pre>

As long as the lspci looks like the above then we should be OK with the existing.

File: tasks/ilo.yml [hunk 3]
```diff
@@ -1,34 +1,33 @@
...
 
-   # We need to get the "iLO4" from the above into a varible
-   # using the python split fuction on the '"' character and the 2nd from last
-   # element of the resulting array. "|lower" is a Jinja2 filter
- - name: Getting ilo version
-   set_fact: ilo_version={{ ilo_info.stdout.split('\"')[-2] |lower}}
+- debug:
+    msg: "{{ ilo_info }}"
 
- - assert: { that: ilo_version is match("ilo4") }
```
Comment (martbhell) [inline tasks/ilo.yml:20]: Why remove this? I like to fail early.

**Review by martbhell — CHANGES_REQUESTED** [review]

**Comment by oscarkraemer** [comment]
Hi @FLiPp3r90 There has been no response from you in the last month, therefor I will close this PR. We would love to have your changes but it is really important that we understand your changes. So if you have the time feel free to reopen this PR and answer the unanswered questions. 

Cheers,
Oscar

**Comment by FLiPp3r90** [comment]
Hi Oscar,

this is okay. In our discussion how to identify the ILO version we decide to live with the old behavior (check pci address).


Von: Oscar Kraemer <notifications@github.com>
Gesendet: Freitag, 8. März 2019 12:45
An: CSCfi/ansible-role-hp-firmware-upgrade <ansible-role-hp-firmware-upgrade@noreply.github.com>
Cc: Krahl, Filip <Filip.Krahl@t-systems.com>; Mention <mention@noreply.github.com>
Betreff: Re: [CSCfi/ansible-role-hp-firmware-upgrade] add ilo support higher than ilo4 (#24)


Hi @FLiPp3r90<https://github.com/FLiPp3r90> There has been no response from you in the last month, therefor I will close this PR. We would love to have your changes but it is really important that we understand your changes. So if you have the time feel free to reopen this PR and answer the unanswered questions.

Cheers,
Oscar

**Pull request closed without merging**